client/deis.py

```diff
@@ -378,6 +378,34 @@ def __init__(self):
         self._settings = Settings()
         self._logger = logging.getLogger(__name__)
 
+    def check_connection(self, controller, ssl_verify):
+        url = urlparse.urljoin(controller, '/v1/')
+        error_message = """
+{} does not appear to be a valid Deis controller.
+Make sure that the Controller URI is correct and the server is running.
+""".format(controller)
+
+        try:
+            response = self._session.get(url, allow_redirects=False,
+                                         verify=ssl_verify)
+        except:
```

Comment:
We should catch a specific exception class here instead and log its message too:
```console
try:
    response = self._session.get(url, allow_redirects=False,
                                 verify=ssl_verify)
except requests.exceptions.ConnectionError as err:
    raise EnvironmentError(error_message + '\n' + err.message)
```
(I tried this code interactively, and that's the error `type` thrown when the host can't be contacted. It wraps the `socket.gaierror`.)

Sorry to make you keep tweaking this PR--this is great stuff!